src/odemis/driver/tfsbc.py

```diff
@@ -0,0 +1,357 @@
+                                           cls=(int, float), unit="m",
+                                           setter=self._setShift)
+
+    def _findDevice(self, ports=None, serialnum=None):
+        """
+        Look for a compatible device. Requires at least one of the arguments ports and serialnum.
+        ports (str): port (e.g. "/dev/ttyUSB0") or pattern for port ("/dev/ttyUSB*"), "/dev/fake" will start the simulator
+        serialnum (str): serial number
+        return (str): the name of the port used
+        raises:
+            ValueError: if no device on the ports with the given serial number is found
+        """
+        # At least one of the arguments ports and serialnum must be specified
+        if not ports and not serialnum:
+            raise ValueError("At least one of the arguments 'ports' and 'serialnum' must be specified.")
+
+        # For debugging purpose
+        if ports == "/dev/fake":
+            return ports
+
+        # If no ports specified, check all available ports
+        if ports:
+            names = list(serial.tools.list_ports.grep(ports))
+        else:
+            names = serial.tools.list_ports.comports()  # search all serial ports
+
+        # Look for serial number if available, otherwise make sure only one port matches the port pattern.
+        if serialnum:
+            for port in names:
+                if serialnum in port.description or serialnum in port.hwid:
+                    # "RS485" is in port.description, .hwid presumably contains serial number, TODO: check this!
+                    return port.device  # Found it!
+            else:
+                raise HwError("Beam controller device with serial number %s not found for port %s. " % (serialnum, names) +
+                              "Check the connection.")
+        else:
+            if len(names) == 1:
+                port = names[0]
+                return port.device
+            elif len(names) > 1:
+                raise HwError("Multiple ports detected for beam controller. Please specify a serial number.")
+            else:
+                raise HwError("Beam controller device not found for port %s. Check the connection." % ports)
+
+    def _openSerialPort(self, port):
+        if self._port == "/dev/fake":
+            return BeamShiftControllerSimulator()
+        else:
+            return ModbusSerialClient(method='rtu', port=port,
+                                      baudrate=BAUDRATE, timeout=TIMEOUT,
+                                      stopbits=STOPBITS, parity=PARITY,
+                                      bytesize=BYTESIZE)
+
+    def _setShift(self, value):
+        """
+        :param value (float, float): x, y shift from the center (in m)
+        """
+        xlower, ylower, xupper, yupper = self._metadata.get(model.MD_CALIB_BEAMSHIFT)
+
+        if None in (xlower, ylower, xupper, yupper):
+            raise ValueError("Cannot set shift, MD_CALIB_BEAMSHIFT metadata not specified.")
```

Comment:
I don't think this works. `dict.get()` would return None, which cannot be mapped to a 4-tuple. You'll not even reach here.
I'd suggest to surround the above line with a try/except KeyError.